tools/build.cmd

```diff
@@ -157,6 +157,21 @@ rem ----------------------------------------------------------------------------
 rem this is setting the cmake path in a quoted way
 set "cmake-root=%build-root%\build"
 
+rem Figure out which CMake generator to use
+if "%VisualStudioVersion%" == "14.0" (
+    set "cmake-generator=Visual Studio 14 2015"
+) else if "%VisualStudioVersion%" == "15.0" (
+    set "cmake-generator=Visual Studio 15 2017"
+)
+if NOT DEFINED cmake-generator (
+    echo ERROR: VisualStudioVersion not defined, or not one of ^(14.0, 15.0^)
+    exit /b 1
+)
+if "%build-platform%" == "x64" (
+    set "cmake-generator=!cmake-generator! Win64"
```

Comment:
Wow. Thanks for the detailed explanation. Sounds like an afterthought of a feature! As part of the Node.js get-off-the-fork work I am switching the Node.js build script to use PowerShell because of batch file parameter parsing weirdness. There will still be a `build_nodejs.cmd` file but that's just a small front-end for the PS script. Coming from batch files, PS now feels like a breath of fresh air!